Edit/storyblok guide error fixes (#2228)
withastro/docs: src/pages/en/guides/cms/storyblok.md

        // Add your components here
      },
      apiOptions: {
        // Choose your Storyblok space region

>         region: 'us', // or 'eu'

Reviewer: [optional] `// or 'eu' (default)`

Reviewer 2: The region is mentioned in a callout before but it doesn't hurt to be repeated. I spent a good 2 hours debugging this small option when I first used Storyblok and that's the reason It is included here.

Author: Agreed. I think it's good to mention both because following the storybook docs, they only suggest adding 'us' if you need it (leave blank for eu).

If you're skimming the article it'd be very easy for an EU user to miss that this has been flipped, even with the which is why I added the comment to help draw attention to this area even if you're quickly copying and pasting.

Finally, to connect the `blogPost` Blok to the `BlogPost` component, add a new property to your components object in your Astro config file. 

- The key is the name of the Blok in Storyblok. In this case, it is `blogPost`.
- The value is the path to the component. In this case, it is `storyblok/BlogPost`.

:::caution

>   The `key` should exactly match your Blok name in Storyblok to be referenced correctly. If these don't match, or you're trying to reference a component that doesn't exist in Storyblok you'll get an error.

Reviewer 3: ```suggestion
  The `key` should exactly match your Blok name in Storyblok to be referenced correctly. If these don't match, or you're trying to reference a component that doesn't exist in Storyblok, you'll get an error.
```

> - The key is the name of the Blok. In this case, it is `blogPost`.

Reviewer 2: ```suggestion
- The key is the name of the Blok in Storyblok. In this case, it is `blogPost`.
```

});
```

> :::caution
>   Your components should exactly match your Storyblok Blok library "technical name" field. If these don't match, or you're trying to reference a component that doesn't exist in Storyblok you'll get an error.
>   :::

Reviewer 2: I think we can move this to line 165. Right after the docs explain what they `key` and `value` are.
```suggestion
:::caution
The `key` should exactly match your Blok name in Storyblok to be referenced correctly. If these don't match, or you're trying to reference a component that doesn't exist in Storyblok you'll get an error.
:::
```

It uses the `useStoryblokApi` hook to fetch all the stories with the content type of `blogPost`. It uses the `version` query parameter to fetch the draft versions of the stories when in development mode and the published versions when building for production.

> We aren't passing any props just yet so `Astro.props` is just used to set up the editor in Storyblok, but we could choose to pass additional props if we wanted.

Reviewer 3: ```suggestion
`Astro.props` is used to set up the editor in Storyblok. Additional props can also be passed to your component here, if needed.
```

        slug: story.full_slug === 'home' ? undefined : story.full_slug
      },
      props: {
        story
      }
    }
  })
  return pages
}

const { story } = Astro.props
---
<html lang="en">
  <head>
    <title>Storyblok & Astro</title>
  </head>
  <body>

>     {story.full_slug === 'home' && (

Reviewer 2: For the reader's benefit, I think we should keep the slug page as simple as possible. In this case, what do you think about removing the h1 element but keeping the story to be passed as props? Then the reader can use all the metadata stored in the story to build their page. 🎉

Author: When I was building this tutorial, and debugging some of the issues above, I was struggling to see the difference between the homepage and the blog posts

As they both use the same file, it wasn't clear which I was looking at at any given time, or which was causing errors.

This isn't a problem with storyblok or astro really, the usage of `[...slug]` is correct, but I think that being a little redundant here can actually help people getting started if they have any problems.

If we want to handhold their mistakes, I think the title is helpful, but if we don't then we could let them figure out this debugging step themselves.

    <title>Storyblok & Astro</title>
  </head>
  <body>

>     {story.full_slug === 'home' && (

Reviewer 2: Referencing my previous comment. I think we can replace the content variable with a story variable. So the `<StoryblokComponent blok={story.content} />` will match in both examples and the reader have access to the story metadata.
